### PR TITLE
fix(a11y): Fix a11y bug for error showing up as tooltip

### PR DESCRIPTION
**Pull request by jeeyi**
When text input validation fails, the error message is set to have role="alert".
This may interrupts the screen reader from reading the focused element.

### CODE REVIEW

File: src/components/tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -231,6 +231,9 @@ class Tooltip extends React.Component<TooltipProps, State> {
 
         if (showTooltip) {
             componentProps['aria-describedby'] = this.tooltipID;
+            if (theme === TooltipTheme.ERROR) {
+                componentProps['aria-errormessage'] = this.tooltipID;
```
Comment (tjgupta) [inline src/components/tooltip/Tooltip.tsx]: Hmm, so wouldn't this still need to be on the input? The input is the source element that is in the error state, not the tooltip itself, and with this change now this tooltipID contains the target of the `aria-errormessage`?

Comment (jeeyi) [reply src/components/tooltip/Tooltip.tsx]: Yes. `aria-errormessage` along with `aria-describedby` is on the input. It references the tooltip that contains the actual error message text `Field Required.`. 

Originally, it was set to the `errorMessageID` from `TextInput.js` which had the same error message text. Now that this is removed from `TextInput.js`, I changed it to set it from the `Tooltip.tsx` where I have the tooltip ID.

File: src/components/tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -266,7 +269,7 @@ class Tooltip extends React.Component<TooltipProps, State> {
             <TetherComponent ref={this.tetherRef} {...tetherProps}>
                 {React.cloneElement(React.Children.only(children) as React.ReactElement, componentProps)}
                 {showTooltip && (
-                    <div className={classes} id={this.tooltipID} role="tooltip">
+                    <div className={classes} id={this.tooltipID} role="tooltip" aria-live="polite">
```
Comment (tjgupta) [inline src/components/tooltip/Tooltip.tsx]: Technically I don't think `aria-live` is supposed to be used on elements that are dynamically added/removed from the DOM. 

From MDN (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions): 
"Dynamically adding an element with an aria-live attribute or specialized role to the document also won't result in any announcement by assistive technologies (as, at that point, the browser/assistive technologies are not aware of the live region yet, so cannot monitor it for changes).

Always make sure that the live region is present in the document first, and only then dynamically add/change any content."

But if this makes it read it out, then 🤷‍♂ ...

File: src/components/tooltip/Tooltip.tsx (hunk 1 above)
Comment (tjgupta) [reply src/components/tooltip/Tooltip.tsx]: Ah yep I see now. Misunderstood where `componentProps` were getting used.

File: src/components/tooltip/Tooltip.tsx (hunk 2 above)
Comment (jeeyi) [reply src/components/tooltip/Tooltip.tsx]: I tested without `aria-live="polite"` on the tooltip and it fails ChromeVox from reading the tooltip. NVDA and JAWS work without `aria-live="polite"`.

**Pull request merged**